tests/integration/specification_model_generator_test.py

```diff
@@ -916,5 +916,34 @@ def test_message_field_conditions_provability(tmp_path: Path) -> None:
             end message
                with Byte_Order => Low_Order_First;
       end Test;
-   """
+    """
+    utils.assert_provable_code_string(spec, tmp_path)
+
+
+@pytest.mark.verification
+def test_parameterized_message_set_scalar(tmp_path: Path) -> None:
+    spec = """
+      package Test is
+
+         type Length_16 is mod 2 ** 16;
+
+         type Signature_Length is range 0 .. 512 with Size => 16;
+
+         type Measurements_Response (Signature_Length : Signature_Length;
+                                     Has_Signature    : Boolean) is
+            message
+               Opaque_Length : Length_16;
+               Opaque_Data : Opaque
+                  with Size => 8 * Opaque_Length
+                  then Signature
+                     if Has_Signature = True
+                  then null
+                     if Has_Signature = False;
+               Signature : Opaque
+                  with Size => 8 * Signature_Length;
+            end message
+               with Byte_Order => Low_Order_First;
+
+      end Test;
+    """
     utils.assert_provable_code_string(spec, tmp_path)
```

Comment:
Restricting the to be proven units could significantly improve the proof time (in `test_parameterized_message_set_scalar` as well as `test_message_field_conditions_provability`). Without the `units` argument, also the `RFLX_Types` is proved each time. This is at least the case when no Memcached is used. I guess with Memcached an already proved file with the same content would be detected, right?

Reply:
Done.
Yes, all the rflx-rflx_types files (for example) are identical, so memcached should speed up those proofs.